### ISSUE TITLE
Parsing WebRoot incorrectly after DefinitionsList refactor

### ISSUE THREAD

**Issue by amalagaura** [vanilla/wordpress-vanilla#1]
I see that your validation function is looking for

https://github.com/vanilla/wordpress-vanilla/blob/master/functions.php#L181

`WebRoot" value="` but the latest Vanilla that I installed has the following:

`'WebRoot' : "http:\/\/forum.sitename.com\/"`

Unfortunately I had to hack the plugin to hardcode my site. Looks like the regex needs to be updated. The embedding works after that.

**Comment by johnmontfx** [comment]
I found the same thing. Modified the code to accept the new javascript as well as strip the slashes. Here's what I ended up with:

```
function vf_validate_url($url) {
  $html = vf_rest($url);
  $wr_pos = strpos($html, '\'WebRoot\' : "');
  if ($wr_pos > 0) {
     $webroot = substr($html, $wr_pos + 13);
     $webroot = substr($webroot, 0, strpos($webroot, '"'));
     $webroot = stripslashes($webroot);
     return $webroot;
  } else {
     return FALSE;
  }
}
```

**Comment by galenbrazell** [comment]
This fixed my issue, as well. The plugin needs to be updated. Thanks @johnmontfx

**Comment by linc** [comment]
Thanks for highlighting this, and for a proposed fix, @johnmontfx. Much appreciated.

**Comment by DanThomas64** [comment]
Could somebody please explain the fix to me? I tried copying the code from johnmontfx into the vanilla-forums/functions.php file but it made no difference. The instructions for the set up of this plugin are somewhat unclear so I can't be sure if the fix isn't working or my configuration is wrong. 

Any help would be greatly appreciated.

**Comment by galenbrazell** [comment]
Been waiting on an official fix/update for this myself, but your mistake is that you need to edit the wordpress plugin itself with johnmontfx's code. Not your forums functions.php.

Found at  wp-content/plugins/vanilla-forums/functions.php

Hope that helps.

**Comment by DanThomas64** [comment]
Sorry maybe that was unclear. That is the file that I was editing. Full path:- /public_html/iogc.co.uk/wp-content/plugins/vanilla-forums

I have seen that there are a bunch of problems with the GoDaddy hosting when running this plugin and have tried a few work-a-rounds with that. I fear that the combined changes that I have made are probably making the problems worse. Truth be told I'm not much of a coder with only very basic knowledge of what I'm doing. Very frustrating.

Thanks for the help though.

**Comment by DanThomas64** [comment]
Still no update on this then. Bit dissapointing.

**Comment by keegantennant** [comment]
@johnmontfx Just made a github account to thank you for your code snippet. It fixed my problem! Turns out the host I'm with has a type of loopback disabled that caused it all. But your code made things work. +1 internet to you.

**Comment by linc** [comment]
Closed by #3. This has been updated on wordpress.org. Thanks everyone.
